Library/Formula/camlp5.rb

```diff
@@ -13,20 +13,41 @@ class Camlp5 < Formula
 
   deprecated_option "strict" => "with-strict"
   option "with-strict", "Compile in strict mode"
+  option "with-doc", "Install the html and info documentation"
+  option "with-tex", "Install the pdf, ps, and tex documentation"
 
   depends_on "ocaml"
+  depends_on "gnu-sed" => :build if build.with?("doc") || build.with?("tex")
+  depends_on :tex => [:build, :optional]
+  depends_on "ghostscript" => :build if build.with?("tex") && !File.exist?("/usr/local/bin/ps2pdf")
 
   def install
-    if build.with? "strict"
-      strictness = "-strict"
-    else
-      strictness = "-transitional"
+    if build.with?("tex")
+      ps2pdf = which("/usr/local/bin/ps2pdf") # known MacTex installation path
+      ps2pdf ||= which(Formula["ghostscript"].bin/"ps2pdf")
+      odie "Sorry can't find your ps2pdf from MacTex's or Homebrew's GhostScript" if ps2pdf.nil?
```

Comment:
```
$ brew install --with-tex --build-bottle --with-doc camlp5
Error: You must `brew link ghostscript` before camlp5 can be installed
```
Odd behavior. If I preinstall brew's ghostscript, and ignore the link error at ghostscript-install time due to the presence of the MacTeX ghostscript, then brew refuses to install camlp5 even though ghostscript is only a build time dependency with the error above. On the other hand, if I do not preinstall ghostscript, it doesn't balk and successfully installs and uses brew ghostscript during the build despite the presence of the MacTeX ghostscript. This strikes me as a bug since there's no reason it couldn't do the same thing in both cases regardless of the brew link step's success or failure at ghostscript-install time.